lib/oauther.ex

```diff
@@ -60,75 +81,91 @@ defmodule OAuther do
     key <> "=\"" <> value <> "\""
   end
 
+  @spec compose_key(Credentials.t) :: String.t
   defp compose_key(creds) do
     [creds.consumer_secret, creds.token_secret]
     |> Enum.map_join("&", &percent_encode/1)
   end
 
+  @spec read_private_key(Path.t) :: term
   defp read_private_key(path) do
     File.read!(path)
     |> :public_key.pem_decode
     |> hd() |> :public_key.pem_entry_decode
   end
 
+  @spec base_string(String.t, String.t, params) :: String.t
   defp base_string(verb, url, params) do
     {uri, query_params} = parse_url(url)
     [verb, uri, params ++ query_params]
     |> Stream.map(&normalize/1)
     |> Enum.map_join("&", &percent_encode/1)
   end
 
+  @spec normalize(String.t) :: String.t
   defp normalize(verb) when is_binary(verb),
     do: String.upcase(verb)
 
+  @spec normalize(URI.t) :: URI.t
   defp normalize(%URI{host: host} = uri),
     do: %{uri | host: String.downcase(host)}
 
+  @spec normalize(nonempty_list(param)) :: String.t
   defp normalize([_ | _] = params) do
     Enum.map(params, &percent_encode/1)
     |> Enum.sort
     |> Enum.map_join("&", &normalize_pair/1)
   end
 
+  @spec normalize_pair(pair_of_strings) :: String.t
   defp normalize_pair({key, value}) do
     key <> "=" <> value
   end
 
+  @spec parse_url(String.t) :: {URI.t, [pair_of_strings]}
   defp parse_url(url) do
     uri = URI.parse(url)
     {%{uri | query: nil}, parse_query_params(uri.query)}
   end
 
+  @spec parse_query_params(nil) :: []
   def parse_query_params(nil), do: []
 
+  @spec parse_query_params(params) :: [pair_of_strings]
   def parse_query_params(params) do
     URI.query_decoder(params)
     |> Enum.into([])
   end
 
+  @spec nonce() :: binary
```

Comment:
Indeed because they don't need to. However, it doesn't harm to spec them too if it helps any other developer to understand the code. After all, this is why I started spec'ing your library.
Do you still want me to remove the private ones ?